r1cs/gadgets/std/src/pairing/mnt6/mod.rs

```diff
@@ -126,15 +129,26 @@ impl<P: MNT6Parameters> PairingGadget<MNT6p<P>, P::Fp> for MNT6PairingGadget<P>
         };
 
         //Final exp last chunk
-
         let elt_q = elt.clone()
             .frobenius_map(cs.ns(|| "elt_q_frobenius_1"), 1)?;
 
         let w1_part = elt_q
             .cyclotomic_exp(cs.ns(|| "compute w1"), P::FINAL_EXPONENT_LAST_CHUNK_1)?;
 
-        let w0_part = elt.clone()
-            .cyclotomic_exp(cs.ns(|| "compute w0"),P::FINAL_EXPONENT_LAST_CHUNK_ABS_OF_W0)?;
+        let w0_part = if P::FINAL_EXPONENT_LAST_CHUNK_W0_IS_NEG {
+            // in this case we need the inverse of elt
+            let elt_inv = {
+                let elt_inv_q3_over_elt_inv = value_inv.clone()
```

Comment:
Can you check if the .clone() is needed here ?

Reply:
It's not needed. I checked that via cargo test and pushed the commit.